src/classes/PS_CommitmentRequest.cls

```diff
@@ -239,20 +265,40 @@ public inherited sharing class PS_CommitmentRequest {
             setSchedules(reqBody, scheduleService.buildNewSchedules(rd), currencyCode);
         }
 
-
-
-
         if (isElevateCampaignChanged(rd, oldRd)) {
-            ProductMetadata productMetadata = new ProductMetadata()
-                .withCampaign(rd.npe03__Recurring_Donation_Campaign__c);
-            
-            reqBody
-                .withProductMetadata(productMetadata);
+            Map<String, Object> updatedMetadata = buildUpdatedProductMetadata(rd);
+            reqBody.withProductMetadata(updatedMetadata);
         }
 
         return reqBody;
     }
 
+    private Map<String, Object> buildUpdatedProductMetadata(npe03__Recurring_Donation__c rd) {
+        Map<String, Object> existingProductMetadata = getExistingProductMetadata(rd.CommitmentId__c);
+        ProductMetadata productMetadata = new ProductMetadata()
+                .withCampaign(rd.npe03__Recurring_Donation_Campaign__c);
+
+        Map<String, Object> mergedMetadata = productMetadata.mergeWithExistingMetadata(existingProductMetadata);
+        return mergedMetadata;
+    }
+
+    private Map<String, Object> getExistingProductMetadata(String commitmentId) {
+        Map<String, Object> responseBody = getElevateCommitment(commitmentId);
+
+        Map<String, Object> commitment = (Map<String, Object>) responseBody.get('commitment');
+        Map<String, Object> productMetadata = (Map<String, Object>) commitment.get('productMetadata');
+
+        return productMetadata;
+    }
+
+    private Map<String, Object> getElevateCommitment(String commitmentId) {
+        HttpRequest commitmentGetRequest = buildGetRequest(commitmentId);
+        UTIL_Http.Response response = requestService.sendRequest(commitmentGetRequest);
```

Comment:
Not sure where to add it, but I'd feel better if this was in a try/catch 😬

Reply:
The same try/catch that's in the update call would catch any errors from this callout, which I think is what we'd want. If the GET request times out, we'd want the same handling as a PUT/POST timeout. 
https://github.com/SalesforceFoundation/NPSP/blob/41a6bcd22c75e9c720cbfb4cf6b6d04ca1c1792b/src/classes/RD2_EntryFormController.cls#L278

Reply:
Oh, excellent, I should've looked further upstream 👍